### ISSUE TITLE
[Quest][3.3.5a] Gerenzo's Orders Fix

### ISSUE THREAD

**Issue by TerrorBringer** [TrinityCore/TrinityCore#18588]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

Here's the fix for quest http://www.wowhead.com/quest=1090/gerenzos-orders which I ported from SunwellCore (https://github.com/Kittnz/Sunwell) and fixed the errors. All credits goes to everyone involved in SunwellCore and its contributors.

<details>

 ```sql
-- Gerenzo's Orders (1090)

UPDATE creature_template SET AIName='SmartAI', ScriptName='' WHERE entry=4276;

SET @ENTRY := 4276;
DELETE FROM creature_text WHERE entry=@Entry;
INSERT INTO creature_text VALUES
(@Entry, 0, 0, 'Watch out for the Kobolds!', 12, 0, 100, 0, 0, 0, 0, 0, 'Piznik'),
(@Entry, 1, 0, "I'm getting close!", 12, 0, 100, 0, 0, 0, 0, 0, 'Piznik'),
(@Entry, 2, 0, "Yay! I have found it!", 12, 0, 100, 0, 0, 0, 0, 0, 'Piznik');

SET @ENTRY := 4276;
DELETE FROM `smart_scripts` WHERE (`entryorguid`=@Entry AND `source_type`=0);
DELETE FROM smart_scripts WHERE entryorguid=@Entry*100 AND source_type=9;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(@Entry, 0, 0, 0, 19, 0, 100, 0, 1090, 0, 0, 0, 80, @Entry*100, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "On Quest Accept - Script9"),
(@Entry*100, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 19, 512, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Script9 - Remove NPC Immune Flag'),
(@Entry*100, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 81, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Script9 - Remove NPC Flags'),
(@Entry*100, 9, 2, 0, 0, 0, 100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Script9 - Talk'),
(@Entry*100, 9, 3, 0, 0, 0, 100, 0, 5000, 5000, 0, 0, 12, 3998, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 956.209, -260.946, -4.515244, 6.022552, 'Script9 - Summon creature'),
(@Entry*100, 9, 4, 0, 0, 0, 100, 0, 0, 0, 0, 0, 12, 4001, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 957.395, -258.515, -4.177891, 5.508116, 'Script9 - Summon creature'),
(@Entry*100, 9, 5, 0, 0, 0, 100, 0, 15000, 15000, 0, 0, 12, 3998, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 956.209, -260.946, -4.515244, 6.022552, 'Script9 - Summon creature'),
(@Entry*100, 9, 6, 0, 0, 0, 100, 0, 0, 0, 0, 0, 12, 4001, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 957.395, -258.515, -4.177891, 5.508116, 'Script9 - Summon creature'),
(@Entry*100, 9, 7, 0, 0, 0, 100, 0, 25000, 25000, 0, 0, 12, 3998, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 956.209, -260.946, -4.515244, 6.022552, 'Script9 - Summon creature'),
(@Entry*100, 9, 8, 0, 0, 0, 100, 0, 0, 0, 0, 0, 12, 4001, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 957.395, -258.515, -4.177891, 5.508116, 'Script9 - Summon creature'),
(@Entry*100, 9, 9, 0, 0, 0, 100, 0, 25000, 25000, 0, 0, 12, 3998, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 956.209, -260.946, -4.515244, 6.022552, 'Script9 - Summon creature'),
(@Entry*100, 9, 10, 0, 0, 0, 100, 0, 0, 0, 0, 0, 12, 4001, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 957.395, -258.515, -4.177891, 5.508116, 'Script9 - Summon creature'),
(@Entry*100, 9, 11, 0, 0, 0, 100, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Script9 - Talk'),
(@Entry*100, 9, 12, 0, 0, 0, 100, 0, 25000, 25000, 0, 0, 12, 3998, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 956.209, -260.946, -4.515244, 6.022552, 'Script9 - Summon creature'),
(@Entry*100, 9, 13, 0, 0, 0, 100, 0, 0, 0, 0, 0, 12, 4001, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 957.395, -258.515, -4.177891, 5.508116, 'Script9 - Summon creature'),
(@Entry*100, 9, 14, 0, 0, 0, 100, 0, 25000, 25000, 0, 0, 12, 3998, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 956.209, -260.946, -4.515244, 6.022552, 'Script9 - Summon creature'),
(@Entry*100, 9, 15, 0, 0, 0, 100, 0, 0, 0, 0, 0, 12, 4001, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 957.395, -258.515, -4.177891, 5.508116, 'Script9 - Summon creature'),
(@Entry*100, 9, 16, 0, 0, 0, 100, 0, 25000, 25000, 0, 0, 12, 3998, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 956.209, -260.946, -4.515244, 6.022552, 'Script9 - Summon creature'),
(@Entry*100, 9, 17, 0, 0, 0, 100, 0, 0, 0, 0, 0, 12, 4001, 4, 30000, 0, 1, 0, 8, 0, 0, 0, 957.395, -258.515, -4.177891, 5.508116, 'Script9 - Summon creature'),
(@Entry*100, 9, 18, 0, 0, 0, 100, 0, 20000, 20000, 0, 0, 1, 2, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Script9 - Talk'),
(@Entry*100, 9, 19, 0, 0, 0, 100, 0, 0, 0, 0, 0, 26, 1090, 0, 0, 0, 0, 0, 18, 40, 0, 0, 0, 0, 0, 0, 'Script9 - Area Explored'),
(@Entry*100, 9, 20, 0, 0, 0, 100, 0, 0, 0, 0, 0, 18, 512, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Script9 - Set NPC Immune Flag'),
(@Entry*100, 9, 21, 0, 0, 0, 100, 0, 0, 0, 0, 0, 81, 2, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Script9 - Set NPC Flags');
```

</details>

Win 8.1 x64
90a5811
[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by Keader** [comment]
https://github.com/TrinityCore/TrinityCore/issues/18586#issuecomment-269141107

**Comment by TerrorBringer** [comment]
Updated, I will update the others aswell

**Comment by offl** [comment]
This was fixed in 97e70b59.
Also seems like these texts are guessed.
```sql
(@Entry, 0, 0, 'Watch out for the Kobolds!', 12, 0, 100, 0, 0, 0, 0, 0, 'Piznik'),
(@Entry, 1, 0, "I'm getting close!", 12, 0, 100, 0, 0, 0, 0, 0, 'Piznik'),
(@Entry, 2, 0, "Yay! I have found it!", 12, 0, 100, 0, 0, 0, 0, 0, 'Piznik');
```
